File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/lookup/LookupServers.java
package com.lasthopesoftware.bluewater.client.connection.builder.lookup;

import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library;
import com.namehillsoftware.handoff.promises.Promise;

public interface LookupServers {
	Promise<ServerInfo> promiseServerInformation(Library library);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/connection/builder/lookup/LookupServers.kt
package com.lasthopesoftware.bluewater.client.connection.builder.lookup

import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise

interface LookupServers {
	fun promiseServerInformation(libraryId: LibraryId): Promise<ServerInfo?>
}
